libgraphics/include/pivot/graphics/abstract/AImmediateCommand.hxx

```diff
@@ -36,16 +36,17 @@ public:
 
     template <typename T>
     /// Copy the source buffer into the destination
-    void copyBuffer(AllocatedBuffer<T> &src, AllocatedBuffer<T> &dst)
+    void copyBuffer(AllocatedBuffer<T> &src, AllocatedBuffer<T> &dst, vk::DeviceSize srcOffset = 0,
+                    vk::DeviceSize dstOffset = 0)
     {
         if (src.getBytesSize() > dst.getAllocatedSize())
```

Comment:
Check with the offsets too. Otherwise you can write past the end of the dst buffer.
```suggestion
        if (src.getBytesSize() - srcOffeset > dst.getAllocatedSize() - dstOffset)
```